File: qlty-cli/tests/lang/cpp/basic.in/Returns.cpp
#include <string>

class Returns {
public:
    void f0() {
    }

    void f1() {
        return;
    }

    std::string f2(int x) {
        if (x == 1) {
            return "one";
        } else if (x == 2) {
            return "two";
        } else if (x == 3) {
            return "three";
        } else if (x == 4) {
            return "four";
        } else if (x == 5) {
            return "five";
        } else if (x == 6) {
            return "six";
        } else if (x == 7) {
            return "seven";
        }
        return "other";
    }
};

File: qlty-cli/tests/lang/elixir/basic.in/Returns.ex
# Elixir has no `return`: this branchy module must report no return-statements issues.
defmodule Returns do
  def classify(value) do
    case value do
      0 -> :zero
      1 -> :one
      2 -> :two
      3 -> :three
      4 -> :four
      5 -> :five
      _ -> :many
    end
  end

  def describe(value) do
    cond do
      value < 0 -> :negative
      value == 0 -> :zero
      value < 10 -> :small
      value < 100 -> :medium
      value < 1000 -> :large
      true -> :huge
    end
  end
end
